ibis/backends/sql/rewrites.py

```diff
@@ -236,9 +236,14 @@ def sqlize(
     simplified = sqlized.replace(merge_select_select)
 
     # extract common table expressions while wrapping them in a CTE node
-    ctes = extract_ctes(simplified)
-    subs = {cte: CTE(cte) for cte in ctes}
-    result = simplified.replace(subs)
+    ctes = set(extract_ctes(simplified))
+
+    def wrap(node, _, **kwargs):
+        new = node.__recreate__(kwargs)
+        return CTE(new) if node in ctes else new
+
+    result = simplified.replace(wrap)
+    ctes = reversed([cte.parent for cte in result.find(CTE)])
```

Comment:
Is the `find` + `reversed` necessary to preserve topological order? If so, can you add a short comment about it here?

Reply:
Find traverse in DFS/BFS not topological. Replace do traverse in topological order so we could extract the rewritten CTEs in the closure, but my first naive attempt didn't work and I was lazy inspecting it. We can defer it to a follow-up.

Reply:
Ok, I will look through the snapshots again to make sure CTEs are occurring topological order.